### PR TITLE
vere: adds versions to all pier artifacts

### PR DESCRIPTION
**Pull request by joemfb**
... including the checkpoint, checkpoint patch files, the event log, and jammed snapshots ("rock").

Additionally, this PR updates the format of the event log, both metadata and events. The metadata had an unnecessary layer of jamming, while the events had an unnecessary layer of cell allocation and a misalignment of events and the mug of the resulting kernel (both artifacts of the old IPC message format).

In cleaning up the persistent `u3_arvo` struct, I had to decide what to do with the i/o drivers' use of the "instance number" therein, as some use the instance number to create unique wires per process. While I suspect that this is entirely unnecessary, I'm not quite sure, and it's an easy thing to change in the future. I've implemented equivalent behavior with per-driver values rather than per-process.

This PR is a draft because the checkpoint version number is not yet useful -- it's the last member of the `u3_home` struct, which makes it quite useless for format discrimination. There's pointer math in the loom initialization, and I'm too tired to track down where it's gone wrong (but I suspect #979 maybe related); I'll revisit tomorrow.

@eglaysher, the event log format changes will need to be implemented in king-haskell as well.

### CODE REVIEW

**Comment by eglaysher** [comment]
A question about the changes to lmdb here: is this just adding a version key/value pair to the meta table? I want to make sure I'm not missing something.

**Comment by joemfb** [comment]
No, the key change is to the structure of the persisted events themselves.

Before, log entries were jammed cells, where the head was the mug of the kernel from the "previous" event, and the tail was the "current event".

Now, log entries are byte buffers, where the first 4 bytes are the mug of the kernel from the "current" event, and the remaining bytes are the jammed current event.

**Comment by Fang-** [comment]
Changes so far look fine. I'm not very familiar with this part of vere, but nothing jumps out as weird, and the cleanup is good. Withholding approval until no longer WIP!

**Comment by eglaysher** [comment]
I have a draft of the new format in KH (https://github.com/urbit/urbit/tree/eg/hs-vere-versions), but it looks like I'm having issues with the urbit-worker being unstable. Is that what you were talking about with regards to loom math?

**Comment by joemfb** [comment]
No, this should be working in its current state -- I didn't push the failing changes. Do you have any more details about what you're seeing?

**Comment by joemfb** [comment]
I have no idea why github is showing so many commits; here's the real diff: https://github.com/urbit/urbit/compare/b6edb8a4c...2edfc51?expand=1

**Comment by matildepark** [comment]
Wouldn't merging this duplicate those commits? Wondering if a force push changed the tree somewhere in there.

**Comment by joemfb** [comment]
I think it's just that the Ui state is broken: 
https://github.com/urbit/urbit/compare/na-release/candidate...jb/vere-versions?expand=1

**Comment by joemfb** [comment]
"have you tried turning it off an on again?"

File: pkg/urbit/include/noun/vortex.h [hunk 1]
```diff
@@ -5,22 +5,22 @@
   /**  Data structures.
   **/
     /* u3v_arvo: modern arvo structure.
+    **       NB: packed to perserve word alignment given [eve_d]
     */
-      typedef struct _u3v_arvo {
-        c3_d    ent_d;                    //  event number
+      typedef struct __attribute__((__packed__)) _u3v_arvo {
+        c3_d  eve_d;                      //  event number
```
Comment (philipcmonk) [inline pkg/urbit/include/noun/vortex.h:11]: Press F for "bumping ent_d"

File: pkg/urbit/vere/io/ames.c [hunk 2]
```diff
@@ -1318,10 +1319,14 @@ _ames_io_talk(u3_auto* car_u)
   u3_ames* sam_u = (u3_ames*)car_u;
   _ames_io_start(sam_u);
 
-  // send born event
+  //  send born event
   //
   {
-    u3_noun wir = u3nt(c3__newt, u3k(u3A->sen), u3_nul);
+    //  XX remove [sev_l]
```
Comment (philipcmonk) [inline pkg/urbit/vere/io/ames.c:1325]: Is this XX for now or later?

File: pkg/urbit/vere/io/http.c [hunk 3]
```diff
@@ -1738,7 +1737,9 @@ _http_serv_start_all(u3_httd* htd_u)
 
     //  XX remove [sen]
     //
-    u3_noun wir = u3nt(u3i_string("http-server"), u3k(u3A->sen), u3_nul);
+    u3_noun wir = u3nt(u3i_string("http-server"),
+                       u3dc("scot", c3__uv, htd_u->sev_l),
```
Comment (philipcmonk) [inline pkg/urbit/vere/io/http.c:1741]: Aren't these various `scot` calls rather expensive?

**Review by philipcmonk — APPROVED** [review]
Couple questions; no objections

File: pkg/urbit/vere/io/ames.c (hunk 2 above)
Comment (joemfb) [reply pkg/urbit/vere/io/ames.c:1325]: It's for later; the question is just: is it desirable for these wires/ducts to change with each process lifetime? I tend to think not, but I'm not sure. (While I'm thinking of it, I should really get rid of `%newt` ...)

File: pkg/urbit/vere/io/http.c (hunk 3 above)
Comment (joemfb) [reply pkg/urbit/vere/io/http.c:1741]: Yeah, they are. Here and above, we should really call them once and store them in the driver/server/connection/request structs as strings. I'll note that for later.

**Pull request merged**